src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala

```diff
@@ -173,11 +179,26 @@ final case class AuthApiServiceImpl(
     queueService
       .send(jwtDetails)
       .void
-      .recoverWith(ex =>
-        Future.successful(
-          logger.error(s"Unable to save JWT details to queue. Details: ${jwtDetails.readableString}", ex)
+      .recoverWith { case sendError =>
+        val jwtPathInfo: JWTPathInfo = generateJWTPathInfo(dateTimeSupplier)
+        val jsonStr: String          = jwtDetails.readableString
+        val content: Array[Byte]     = jsonStr.getBytes()
+
+        logger.error(s"Unable to send JWT to the queue. JWT: $jsonStr", sendError)
+        logger.info(
+          s"Storing JWT to fallback bucket at " +
+            s"$jwtFallbackBucketPath/${jwtPathInfo.path}/${jwtPathInfo.filename}"
         )
-      )
+
+        fileManager
+          .storeBytes(jwtFallbackBucketPath, jwtPathInfo.path, jwtPathInfo.filename)(content)
+          .void
+          .recoverWith { case storeError =>
+            val errorMsg = s"Unable to save JWT details to fallback bucket. JWT: $jsonStr"
+            logger.error(errorMsg, storeError)
+            Future.failed(new RuntimeException(errorMsg, storeError))
```

Comment:
Minor: use ad hoc Exception here instead of RuntimeException

Reply:
> Minor: use ad hoc Exception here instead of RuntimeException

I thought about this but with an hoc Exception I can include only the `errorMsg` and not the `storeError`, but on a second thought we are already logging it so it shouldn't be a problem.
What do you think about this? Should I just go ahead and create the Exception like you suggested since we are already logging the error?